src/seedu/addressbook/data/Printable.java

```diff
@@ -0,0 +1,21 @@
+package seedu.addressbook.data;
+
+/**
+ * Allows classes that implement this interface to be able to produce a printable representation of the object
+ */
+public interface Printable {
+    public static final String PRINTABLE_SEPARATOR = ", ";
+
+    public abstract String getPrintableString();
+
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    static String getPrintableString(Printable... printables) {
```

Comment:
This is an interesting implementation =).
Though it may seem excessive for this activity's requirements.
Furthermore, wouldn't Address, Email and Phone be able to call getPrintableString(...)?
Is this really following SRP?